backend/czi_hosted/data_common/dataset_metadata.py

```diff
@@ -0,0 +1,93 @@
+    #  remain while we support covid 19 cell atlas.
+    #   See ticket https://app.zenhub.com/workspaces/single-cell-5e2a191dad828d52cc78b028/issues/chanzuckerberg/corpora-data-portal/1281 # noqa
+    if server_config.single_dataset__datapath:
+        datapath = server_config.single_dataset__datapath
+        return datapath
+    else:
+        dataset_explorer_location = dataset_explorer_location.split("/")
+        dataset = dataset_explorer_location.pop(-1)
+        url_dataroot = "/".join(dataset_explorer_location)
+        dataroot = None
+        for key, dataroot_dict in server_config.multi_dataset__dataroot.items():
+            if dataroot_dict["base_url"] == url_dataroot:
+                dataroot = dataroot_dict["dataroot"]
+                break
+        if dataroot is None:
+            raise DatasetAccessError(f"Invalid dataset {url_dataroot}/{dataset}")
+        datapath = path_join(dataroot, dataset)
+        # path_join returns a normalized path.  Therefore it is
+        # sufficient to check that the datapath starts with the
+        # dataroot to determine that the datapath is under the dataroot.
+        if not datapath.startswith(dataroot):
+            raise DatasetAccessError(f"Invalid dataset {url_dataroot}/{dataset}")
+        return datapath
+
+
+def get_dataset_metadata_for_explorer_location(dataset_explorer_location: str, app_config: AppConfig):
+    """
+    Given the dataset access location(the path to view the dataset in the explorer including the dataset root,
+    also used as the cache key) and the explorer web base url (from the app_config) this function returns the location
+    of the dataset, along with additional metadata.
+    The dataset location is is either retrieved from the data portal, or built based on the dataroot information stored
+    in the server config.
+    In the case of a single dataset the dataset location is pulled directly from the server_config.
+    """
+    if app_config.server_config.data_locator__api_base:
+        explorer_url_path = f"{app_config.server_config.get_web_base_url()}/{dataset_explorer_location}"
+        dataset_metadata = request_dataset_metadata_from_data_portal(
+            data_portal_api_base=app_config.server_config.data_locator__api_base,
+            explorer_url=explorer_url_path
+        )
+        if dataset_metadata:
+            return dataset_metadata
+    server_config = app_config.server_config
+    dataset_metadata = {
+        "collection_id": None,
+        "collection_visibility": None,
+        "dataset_id": None,
+        "s3_uri": None,
+        "tombstoned": False
+    }
+    dataset_metadata["s3_uri"] = extrapolate_dataset_location_from_config(
+        server_config=server_config, dataset_explorer_location=dataset_explorer_location)
+    if dataset_metadata["s3_uri"] is None:
+        return common_rest.abort_and_log(HTTPStatus.BAD_REQUEST, "Invalid dataset NONE", loglevel=logging.INFO)
```

Comment:
The error message here seems a bit convoluted -- could you make it a bit clearer please?